apps/root/src/hooks/useTokenListByChainId.ts

```diff
@@ -1,46 +1,24 @@
 import React from 'react';
-import { TokenListByChainId, TokenListId } from '@types';
-import toPairs from 'lodash/toPairs';
-import compact from 'lodash/compact';
-import orderBy from 'lodash/orderBy';
-import { DCA_TOKEN_BLACKLIST, TOKEN_BLACKLIST } from '@constants';
-import { getProtocolToken, PROTOCOL_TOKEN_ADDRESS, TOKEN_MAP_SYMBOL } from '@common/mocks/tokens';
-import { useSavedAllTokenLists, useTokensLists } from '@state/token-lists/hooks';
+import { TokenListByChainId } from '@types';
+import useTokenList, { UseTokenListProps } from './useTokenList';
 
-function useTokenListByChainId({ filter = false } = {}) {
-  const tokensLists = useTokensLists();
-  const savedAllTokenLists = useSavedAllTokenLists();
+function useTokenListByChainId(tokenListProps: UseTokenListProps) {
+  const tokenList = useTokenList(tokenListProps);
 
-  const tokenListByChainId: TokenListByChainId = React.useMemo(() => {
-    const filteredLists = orderBy(
-      compact(toPairs(tokensLists).map(([key, list]) => (!filter || savedAllTokenLists.includes(key) ? list : null))),
-      ['priority'],
-      ['desc']
-    );
-
-    return filteredLists.reduce<TokenListByChainId>((acc, tokensList) => {
-      tokensList.tokens.forEach((token) => {
-        if (!filter || !(savedAllTokenLists ? TOKEN_BLACKLIST : DCA_TOKEN_BLACKLIST).includes(token.address)) {
-          const { chainId } = token;
-          if (!acc[chainId]) {
-            // eslint-disable-next-line no-param-reassign
-            acc[chainId] = {
-              [`${chainId}-${PROTOCOL_TOKEN_ADDRESS}`]: getProtocolToken(chainId),
-            };
-          }
-          // eslint-disable-next-line no-param-reassign
-          acc[chainId][`${chainId}-${token.address.toLowerCase()}` as TokenListId] = {
-            ...token,
-            name: TOKEN_MAP_SYMBOL[token.address] || token.name,
-          };
-        }
-      });
-
-      return acc;
-    }, {});
-  }, [tokensLists, savedAllTokenLists, filter]);
-
-  return tokenListByChainId;
+  return React.useMemo(
+    () =>
+      Object.entries(tokenList).reduce<TokenListByChainId>(
+        (acc, [tokenKey, token]) => ({
+          ...acc,
```

Comment:
This could bring down performance, simply by the amount of tokens we may handle. Lets move this function to modify acc directly